Add API to get list of COS objects

jaywcarman opened the pull request:
Related to:
- ManageIQ/manageiq-providers-ibm_cloud#453

TODO: Add specs

kbrock commented:
ok
- build is failing because it is expecting a feature for this
- also, you added "supports => true", so I am expecting IbmCloud::...::CloudObjectStoreObject to have `supports :create`.
- I didn't see the `supports`. Possibly you already added this in an unmerged PR, or very possibly I'm misremembering what `:supports` means in this context.

Thanks Jay for all the great work.

kbrock commented on config/api.yml:
```diff
@@ -608,6 +608,24 @@
+    :resource_actions:
+      :get:
+      - :name: read
+        :identifier: cloud_object_store_obect_show
```
```suggestion
        :identifier: cloud_object_store_object_show
```

jaywcarman commented:
I added some changes to `CloudTemplatesController` earlier that in retrospect would be better off in a separate PR. Just removed it with a force push and opened https://github.com/ManageIQ/manageiq-api/pull/1223.

jaywcarman commented on app/controllers/api/cloud_object_store_objects_controller.rb:
```diff
@@ -0,0 +1,10 @@
+  class CloudObjectStoreObjectsController < BaseProviderController
+    def create_resource(type, _ems_id = nil, data = {})
+      create_ems_resource(type, data, :supports => true) do |ems, klass|
+        resource_search(data["cloud_tenant_id"], :cloud_tenants) if data["cloud_tenant_id"]
```
Original scope of this PR is just read-only... so I just removed the `create_resource` method altogether.

jaywcarman commented:
> * build is failing because it is expecting a feature for this

Took me a while to realize it, but the feature was already there... I just typo'd the identifier 😆
```diff
diff --git a/config/api.yml b/config/api.yml
index f97de0fe..da5563ef 100644
--- a/config/api.yml
+++ b/config/api.yml                                   
@@ -610,7 +610,7 @@            
         :identifier: cloud_object_store_container_show
   :cloud_object_store_objects:            
     :description: Cloud Object Store Objects
-    :identifier: cloud_object_store_objects
+    :identifier: cloud_object_store_object
     :options:                     
     - :collection       
     :verbs: *gp
```

jaywcarman commented:
Existing specs are now passing locally...  just need to add new specs and then this will be out of WIP.

jaywcarman commented:
@kbrock I think this is ready to go now.

I wasn't sure if I should go ahead and add the `:custom_actions:` option (Similar to COS container [here](https://github.com/ManageIQ/manageiq-api/blob/master/config/api.yml#L595). I'm not familiar with this feature, but from reading the docs it looks like this is just generic support for custom buttons. For my purposes I don't need it, but on the other hand it could be nice to go ahead and enable it for someone down the road.
